skfp/fingerprints/atom_pair.py

```diff
@@ -206,14 +215,31 @@ def _calculate_fingerprint(
             countSimulation=self.count_simulation,
         )
         if self.count:
-            X = [
-                gen.GetCountFingerprintAsNumPy(mol, confId=conf_id)
-                for mol, conf_id in zip(X, conf_ids)
-            ]
+            if self.normalize:
+                X = [
+                    self._scale_by_hac(
```

Comment:
I think that adding this as a separate step, after `if self.count: ... else: ...` would be more readable. It's a separate list comprehension then, and a bit more explicit. This is a kind of postprocessing of count values, so I think that it makes sense. Also warning about scaling for bit variant would be then closer to calling `_scale_by_hac()`

Reply:
Minor drawback is that we have to abandon convention to always operate on X variable but I assume it's still worth it.